lib/rspec/buildkite/analytics/session.rb

```diff
@@ -255,23 +260,21 @@ def queue_and_track_result(ident, result_as_hash)
       @idents_mutex.synchronize do
         @unconfirmed_idents[ident] = result_as_hash
 
-        data = {
+        @send_queue << {
           "action" => "record_results",
           "results" => [result_as_hash]
         }
-
-        @send_queue << data
       end
     end
 
-    def remove_unconfirmed_idents(idents)
-      return if idents.empty?
+    def confirm_idents(idents)
+      retransmit_required = @closing
 
       @idents_mutex.synchronize do
         # Remove received idents from unconfirmed_idents
         idents.each { |key| @unconfirmed_idents.delete(key) }
 
-        @logger.write("received confirm for indentifiers: #{idents.join(", ")}")
+        @logger.write("received confirm for indentifiers: #{idents}")
```

Comment:
Why remove the `join(", ")`? I guess separate UUIDs by comma is not much gain from looking at the array itself.